cfme/tests/infrastructure/test_providers.py

```diff
@@ -334,6 +336,53 @@ def test_provider_rhv_create_delete_tls(request, provider, verify_tls):
     prov.wait_for_delete()
 
 
+@pytest.mark.rhv3
+@test_requirements.rhev
+@pytest.mark.meta(automates=[1691109, 1731237])
+@pytest.mark.provider([RHEVMProvider], selector=ONE_PER_VERSION, override=True, scope="function")
+def test_rhv_guest_devices_count(appliance, has_no_providers, provider, request):
```

Comment:
**Question:**
Instead of `has_no_providers` you can use `setup_provider` fixture, which removes existing provider (if any) as well as will create provider for further `_gd_count()` test  before and after provider refresh .